numpy/lib/function_base.py

```diff
@@ -1526,20 +1541,42 @@ def unwrap(p, discont=pi, axis=-1):
     array([ 0.        ,  0.78539816,  1.57079633,  5.49778714,  6.28318531]) # may vary
     >>> np.unwrap(phase)
     array([ 0.        ,  0.78539816,  1.57079633, -0.78539816,  0.        ]) # may vary
-
+    >>> unwrap([0, 1, 2, -1, 0], interval_size=4)
+    array([0., 1., 2., 3., 4.])
+    >>> unwrap([ 1, 2, 3, 4, 5, 6, 1, 2, 3], interval_size=6)
+    array([1., 2., 3., 4., 5., 6., 7., 8., 9.])
+    >>> unwrap([2, 3, 4, 5, 2, 3, 4, 5], interval_size=4)
+    array([2., 3., 4., 5., 6., 7., 8., 9.])
+    >>> phase_deg = np.mod(np.linspace(0,720,19), 360) - 180
+    >>> unwrap(phase_deg, interval_size=360)
+    array([-180., -140., -100.,  -60.,  -20.,   20.,   60.,  100.,  140.,
+        180.,  220.,  260.,  300.,  340.,  380.,  420.,  460.,  500.,
+        540.])
     """
     p = asarray(p)
     nd = p.ndim
     dd = diff(p, axis=axis)
+    offset = 0
+    if (not min_val is None) and (not max_val is None):
+        interval_size = max_val - min_val
+        slice0list = [slice(None, None)]*nd     # full slices
+        slice0list[axis] = 0
+        slice0 = tuple(slice0list)
+        offset_mul = (p[slice0] - min_val)//interval_size
+        slice0list[axis] = None
+        offset = -offset_mul[tuple(slice0list)]*interval_size
+    if discont is None:
+        discont = interval_size/2
     slice1 = [slice(None, None)]*nd     # full slices
     slice1[axis] = slice(1, None)
     slice1 = tuple(slice1)
-    ddmod = mod(dd + pi, 2*pi) - pi
-    _nx.copyto(ddmod, pi, where=(ddmod == -pi) & (dd > 0))
+    ddmod = mod(dd + interval_size/2, interval_size) - interval_size/2
+    _nx.copyto(ddmod, interval_size/2, where=(ddmod == -interval_size/2) & (dd > 0))
```

Comment:
I'm still confused by this code (the original, not your transformation which looks consistent) - at first glance, it converts `[-I/2, I/2)` to `(-I/2, I/2]`, but for reasons I don't understand does so only if `dd > 0`?

Reply:
Ok, I think it's also not directly clear from the argument about the interval boundaries. I made small example:
Including the line:
```py
>>> a=np.array([0,1,3,4,5,6, 3,4])
>>> unwrap(a, min_val=0, max_val=6)
array([0., 1., 3., 4., 5., 6., 3., 4.])
>>> unwrap(a[::-1], min_val=0, max_val=6)
array([4., 3., 6., 5., 4., 3., 1., 0.])
```
So you see that at the unclear point where the jump has the size of `discont` (here 3), the jump is not corrected, only if the jumps is greater than `3`, e.g. for the array `np.array([0,1,3,4,5,6, 2,3,4])`. Now if we comment out the line with `dd > 0`:

```py
>>> a=np.array([0,1,3,4,5,6, 3,4])
>>> unwrap(a, min_val=0, max_val=6)
array([0., 1., 3., 4., 5., 6., 3., 4.])
>>> unwrap(a[::-1], min_val=0, max_val=6)
array([ 4.,  3.,  0., -1., -2., -3., -5., -6.])
```
the jump is not corrected for the increasing array, but it **is** corrected for the decreasing order of the array. In fact, I have to revise my interpretation of the boundary intervals. Without the line `dd >0` all diffs from the interval `[-I/2, I/2)`  would be kept as they are. But with `dd >0`, the interval changes to `[-I/2, I/2]`. So in fact, all jumps with `abs(dd) > I/2` will be corrected, regardless of the sign. If the line with `dd >0` were dropped, this statement is not true anymore. Whether or not a jump of `I/2` were corrected, would depend on the sign.

Reply:
Right, I think I understand - `unwrap` is using the gradient to break ties?

Reply:
Wait, I think this statement is wrong:

> So in fact, all jumps with abs(dd) > I/2 will be corrected, regardless of the sign

Don't you mean...

> all jumps with `abs(dd) < I/2` will be corrected, regardless of the sign

i.e, jumps of `abs(dd) == I/2` are _never_ corrected?

Reply:
```suggestion
    # the above line made `ddmod[abs(dd) == interval_size/2] == -interval_size/2`.
    # correct these such that `ddmod[abs(dd) == interval_size/2] == sign(dd)*interval_size/2`.
    _nx.copyto(ddmod, interval_size/2, where=(ddmod == -interval_size/2) & (dd > 0))
```

Reply:
> What is meant here, is that jumps of exactly 180 degree are not correct.

I think this is what I was trying to say. Do you agree with my suggested comments above?

Reply:
Sorry, I was just referring to the comment you made in the review, not the suggested change in the code (which is also a comment^^). I agree with with your comments in the code!